starter-api/src/main/java/io/micronaut/starter/api/diff/DiffController.java

```diff
@@ -179,49 +177,45 @@ public Publisher<String> diffApp(
             throw new HttpStatusException(HttpStatus.BAD_REQUEST, e.getMessage());
         }
 
-        return diffFlowable(projectGenerator, generatorContext);
+        return diff(projectGenerator, generatorContext);
     }
 
-    private Publisher<String> diffFlowable(ProjectGenerator projectGenerator, GeneratorContext generatorContext) {
-        return Flux.create(emitter -> {
-            try {
-                // empty string so there is at least some content
-                // if there is no difference
-                emitter.next("");
-                featureDiffer.produceDiff(
-                        projectGenerator,
-                        generatorContext,
-                        new ConsoleOutput() {
-                            @Override
-                            public void out(String message) {
-                                emitter.next(message + LINE_SEPARATOR);
-                            }
+    private String diff(ProjectGenerator projectGenerator, GeneratorContext generatorContext) {
```

Comment:
@yawkat @dstepanov have we changed how `Publisher<String>` is managed? I had to change this. The previous code was timing out in the test.

Reply:
could be https://github.com/micronaut-projects/micronaut-core/pull/11107